src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -79,6 +81,29 @@ public String createProfile(BillingProfileRequestModel billingProfileRequest,
             .submit();
     }
 
+    /**
+     * Update billing profile. We make the following checks:
+     * <ul>
+     *     <le>The service must have proper permissions on the google billing account</le>
+     *     <le>The caller must have billing.resourceAssociation.create permission on the google billing account</le>
+     *     <le>The google billing account must be enabled</le>
+     * </ul>
+     *
+     * @param  billingProfileRequest request with changes to billing profile
+     * @param user the user attempting to update the billing profile
+     * @return jobId of the submitted stairway job
+     */
+    public String updateProfile(BillingProfileUpdateModel billingProfileRequest,
+                                AuthenticatedUserRequest user) {
+        iamService.verifyAuthorization(user, IamResourceType.SPEND_PROFILE, billingProfileRequest.getId(),
+            IamAction.UPDATE_BILLING_ACCOUNT);
+
+        String description = String.format("Update billing for profile id '%s'", billingProfileRequest.getId());
+        return jobService
+            .newJob(description, ProfileUpdateFlight.class, billingProfileRequest, user)
```

Comment:
This is probably an existing convention but I find it a bit confusing that this service both updates the actual data (by using the DAO) and creates flights, which in turn call this service to call the DAO.

I would expect a division where one class only creates flights (which wouldn't need a DAO) and one class only modifies the data.